src/Object/Contact.php

```diff
@@ -143,14 +134,13 @@ public static function markForArchival(array $contact)
 	 *
 	 * @see Contact::markForArchival()
 	 *
-	 * @param array $contact
+	 * @param array $contact contact to be unmarked for archival
 	 * @return null
 	 */
 	public static function unmarkForArchival(array $contact)
 	{
-		$r = q(
-			"SELECT `term-date` FROM `contact` WHERE `id` = %d AND (`term-date` > '%s' OR `archive`)", intval($contact['id']), dbesc('1000-00-00 00:00:00')
-		);
+		$condition = array('`id` => ? AND (`term-date` > ? OR `archive`)', $contact[`id`], NULL_DATE);
+		$r = dba::select('contact', array('term-date'), $condition);
```

Comment:
Is this query only done to check if such a row exists? Then you could use this method:
```$exists = dba::exists('contact', $condition);```

This method only returns ```true``` or ```false```, which means that you don't havr to use ```DBM::is_result``` in that case.